### PR TITLE
Chore: Include Transaction Set Name

### PR DESCRIPTION
**Pull request by noahsettersten**
To start learning about EDI/X12 and Rustler, I've added the ability to retrieve the transaction name (see [docs](https://docs.rs/edi/latest/edi/struct.Transaction.html)).

I've also added a `.tool-versions` file and commented out unused code in the test to fix a couple warnings.

---

Questions:
- [ ] It looks like compiled libraries are committed under `priv/native`. Should I add my updated `libedix_parser.so` as well as a new one that was generated for v `0.1.5`?
- [ ] The edi crate refers to a transaction struct, but it's called a "transaction set" here. Is there any background behind the difference in naming?

### CODE REVIEW

**Review by timgremore — COMMENTED** [review]
@noahsettersten Thank you for this! Changes all look good. In response to your questions:

- re `libedix_parser.so`: I haven't figured out how to get CI to run once per release so I've just been adding a 2nd commit with build artifacts. Did you find `mix rustler_precompiled.download Edix.Parser --only-local`?
- re "transaction set": No reason other than "set" felt redundant. But I don't love any of the names. My intention is to match the struct names to the EDI spec as closely as possible so I need to review this. Additionally, prefixing all my modules with `Edix.EdixWhatever` wasn't a great choice - I'd prefer `Edix.Whatever` after having interacted with it this week.

**Comment by noahsettersten** [comment]
Thanks @timgremore! I wasn't aware of that mix task. When I ran it locally, it logged out that it was downloading a NIF from the releases and caching it. Is the intention for it to download precompiled libraries for local use? That would be a good addition to the README.

I've pushed another commit that updates `libedix_parser.so` and adds a new library output for `v0.1.5`.

For naming, I agree that staying as close to the spec (X12?) as possible will be the most intuitive. I'd be open to removing the extra prefix for the modules as well, since they're namespaced within `Edix`.

**Comment by timgremore** [comment]
@noahsettersten I'm sure there is a better way but the checksum file that CI generates needs to find it's way back onto your system [The release flow](https://github.com/philss/rustler_precompiled/blob/main/PRECOMPILATION_GUIDE.md#the-release-flow).

File: test/edix_test.exs [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+    #     functional_group_control_number: "SM",
+    #     application_sender_code: "915792US00",
+    #     application_receiver_code: "8TMW",
+    #     date: "20240429",
+    #     time: "0614",
+    #     group_reference_number: "SM",
+    #     transaction_set_control_number: "SM",
+    #     transaction_sets: transaction_sets
+    #   }
+    # ]
```
Comment (timgremore) [inline test/edix_test.exs]: Might as well delete these comments.

```suggestion
```

**Review by timgremore — APPROVED** [review]
Thank you @noahsettersten!

**Pull request merged**